fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/RecordMetaDataBuilder.java

```diff
@@ -374,10 +378,41 @@ public RecordMetaDataBuilder setRecords(@Nonnull Descriptors.FileDescriptor file
         if (recordsDescriptor != null) {
             throw new MetaDataException("Records already set.");
         }
+        if (localFileDescriptor != null || !explicitDependencies.isEmpty()) {
+            throw new MetaDataException("Invalid arguments set.");
+        }
         loadFromFileDescriptor(fileDescriptor, processExtensionOptions);
         return this;
     }
 
+    /**
+     * Sets the local file descriptor.
+     *
+     * A local meta-data file descriptor may contain newer versions of record types or indexes that follows the
+     *
+     * This method is handy when two record meta-data are compatible (i.e., they follow
+     * <a href="https://foundationdb.github.io/fdb-record-layer/SchemaEvolution.html">Record Layer guidelines on schema evolution</a>),
+     * but their descriptors are not equal; e.g., a statically-generated proto and its serialized version that's stored
+     * in a meta-data store. A record store created off of the meta-data store may not be able to store a record created
+     * by the statically-generated proto file because of descriptors mismatch. Using this method, the builder can use
+     * the correct version of the descriptor that the record was created with.
+     *
+     * WARNING: There is currently no validation done to make sure the local descriptor is compatible with the records
```

Comment:
Needs a paragraph tag to render correctly. Also, "records" to "record's" or "record".